dotnet/src/Experimental/Process.Abstractions/KernelProcessEventsSubscriberInfo.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System;
+using Microsoft.SemanticKernel.Process;
+
+namespace Microsoft.SemanticKernel;
+
+public class KernelProcessEventsSubscriberInfo
+{
+    private readonly Dictionary<string, List<MethodInfo>> _eventHandlers = [];
+    private readonly Dictionary<string, string> _stepEventProcessEventMap = [];
+    private Type? _processEventSubscriberType = null;
+
+    private IServiceProvider? _subscriberServiceProvider = null;
+
+    protected void Subscribe(string eventName, MethodInfo method)
+    {
+        if (this._eventHandlers.TryGetValue(eventName, out List<MethodInfo>? eventHandlers) && eventHandlers != null)
+        {
+            eventHandlers.Add(method);
+        }
+    }
+
+    public void LinkStepEventToProcessEvent(string stepEventId, string processEventId)
+    {
+        this._stepEventProcessEventMap.Add(stepEventId, processEventId);
+        if (!this._eventHandlers.ContainsKey(processEventId))
+        {
+            this._eventHandlers.Add(processEventId, []);
+        }
+    }
+
+    public void TryInvokeProcessEventFromStepMessage(string stepEventId, object? data)
+    {
+        if (this._stepEventProcessEventMap.TryGetValue(stepEventId, out var processEvent) && processEvent != null)
+        {
+            this.InvokeProcessEvent(processEvent, data);
+        }
+    }
+
+    public void InvokeProcessEvent(string eventName, object? data)
+    {
+        if (this._processEventSubscriberType != null && this._eventHandlers.TryGetValue(eventName, out List<MethodInfo>? linkedMethods) && linkedMethods != null)
+        {
+            foreach (var method in linkedMethods)
+            {
+                // TODO-estenori: Avoid creating a new instance every time a function is invoked - create instance once only?
+                var instance = Activator.CreateInstance(this._processEventSubscriberType, [this._subscriberServiceProvider]);
+                method.Invoke(instance, [data]);
+            }
+        }
+    }
```

Comment:
this is only used in LocalProcess, perhaps need to create a `LocalKernelProcessEventsSubscriber` that focuses on execution only